components/brave_new_tab_ui/reducers/new_tab_reducer.tsx

```diff
@@ -62,6 +68,14 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       addToDispatchQueue(() => {
         gridAPI.calculateGridSites(state)
       })
+      // Run side-effects
+      setTimeout(async function () {
```

Comment:
since this hack is the same as what `addToDispatchQueue` does, maybe better re-use it so when we can afford adopting a middleware and refactoring these cases we can spot this case as well

Reply:
I didn't want to add to `addToDispatchQueue` because it's not dispatching an action. But maybe I should just rename `addToDispatchQueue` to `performSideEffect`.

Reply:
done now - renamed all to `performSideEffect`